components/dashboard/src/Setup.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useEffect, useState } from "react";
+import Modal from "./components/Modal";
+import { getGitpodService, gitpodHostUrl } from "./service/service";
+import { GitIntegrationModal } from "./settings/Integrations";
+
+export default function Setup() {
+
+    const [showModal, setShowModal] = useState<boolean>(false);
+
+    useEffect(() => {
+        (async () => {
+            const dynamicAuthProviders = await getGitpodService().server.getOwnAuthProviders();
+            const previous = dynamicAuthProviders.filter(ap => ap.ownerId === "no-user")[0];
+            if (previous) {
+                await getGitpodService().server.deleteOwnAuthProvider({ id: previous.id });
+            }
+        })();
+    }, []);
+
+    const acceptAndContinue = () => {
+        setShowModal(true);
+    }
+
+    const onAuthorize = (payload?: string) => {
+        // run without await, so the integrated closing of new tab isn't blocked
+        (async () => {
+            window.location.href = gitpodHostUrl.asDashboard().toString();
+        })();
+    }
+
+    return <div>
+        {!showModal && (
+            <Modal visible={true} onClose={() => { }} closeable={false}>
+                <h3 className="pb-2">ALLMOST DONE</h3>
+                <div className="border-t border-b border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 py-4">
+                    <p className="pb-4 text-gray-500 text-base">To complete the setup of Gitpod, you need to create a Git Integration.</p>
+
+                    <div className="flex">
+                        <span className="text-gray-500">
+                            By continuing, you agree to our <a className="underline underline-thickness-thin underline-offset-small hover:text-gray-600" target="gitpod-terms" href="https://www.gitpod.io/self-hosted-terms/">terms</a>.
```

Comment:
**suggestion**: Minor rephrasing to match the one in our login page at Gitpod.io.

```suggestion
                            By using Gitpod, you agree to our <a className="underline underline-thickness-thin underline-offset-small hover:text-gray-600" target="gitpod-terms" href="https://www.gitpod.io/self-hosted-terms/">terms of service</a>.
```

Reply:
ok. 

but why is this a service in case of self-hosted?

Reply:
Ah, good point! I think this refers to the Gitpod as a service being installed. However, I could be wrong. Feel free to leave this as is if you think this could create more confusion.

Reply:
FWIW, here are the [previous self hosted terms](https://github.com/gitpod-io/retired-gatsby-website/blob/master/src/pages/self-hosted-terms.tsx). 📖